improve time window to become cron expression based scheduler (#351)
bottlerocket-os/bottlerocket-update-operator: README.md

              value: "1"
```

> #### Set an Update Time Window - DEPRECATED
> Note!! that these settings are deprecated and will be removed in a future release. You should use the Scheduler settings below.  If you still decide to use these settings, please use "hour:00:00" format only instead of "HH:MM:SS".

Reviewer: Match the code style used elsewhere in this document.
```suggestion
#### Set an Update Time Window - DEPRECATED

Note!! that these settings are deprecated and will be removed in a future release.
You should use the Scheduler settings below.
If you still decide to use these settings, please use "hour:00:00" format only instead of "HH:MM:SS".
```

              value: "1"
```

#### Set an Update Time Window - DEPRECATED
Note!! that these settings are deprecated and will be removed in a future release. You should use the Scheduler settings below.

> 

Reviewer 2: I think we should also update this documentation to reflect that users should use `HOUR:0:0` since the granularity of minutes and seconds are going away.

                fieldRef:
                  fieldPath: spec.nodeName
            - name: UPDATE_WINDOW_START
              value: "09:00:00"
            - name: UPDATE_WINDOW_STOP
              value: "21:00:00"
```

#### Set scheduler
`SCHEDULER_CRON_EXPRESSION` can be used to specify the scheduler in which updates are permitted.
When `SCHEDULER_CRON_EXPRESSION` is "* * * * * * *" (default), the feature is disabled.

> To enable this feature, go to `bottlerocket-update-operator.yaml` and change `SCHEDULER_CRON_EXPRESSION` to a valid cron expression.

Reviewer: This instruction will change once helm is merged?

Author: Yeah, I think so. Maybe @jpmcb has more context on this.

Reviewer 2: Yes - I'm planning to rebase all this on my PR once this gets in 😄

              value: "21:0:0"
```

#### Set scheduler
`SCHEDULER_CRON_EXPRESSION` can be used to specify the scheduler in which updates are permitted.
When `SCHEDULER_CRON_EXPRESSION` is "* * * * * * *" (default), the feature is disabled.

> To enable this feature, go to `bottlerocket-update-operator.yaml` and change `SCHEDULER_CRON_EXPRESSION` to a valid cron expression. A cron expression can be configured to a time window or a specific trigger time. When users specify cron expressions as a time window, the bottlerocket update operator will operate node updates within that update time window. When users specify cron expression as a specific trigger time, brupop will update and complete all waitingForUpdate nodes on the cluster when that time occurs.

Reviewer: My attempt at a clarity in the last sentence.

```suggestion
To enable this feature, go to `bottlerocket-update-operator.yaml` and change `SCHEDULER_CRON_EXPRESSION` to a valid cron expression.
A cron expression can be configured to a time window or a specific trigger time.
When users specify cron expressions as a time window, the bottlerocket update operator will operate node updates within that update time window.
When users specify cron expression as a specific trigger time, brupop will start at the specified time and complete all node updates.
```

> Note: brupop uses UTC (24-hour time notation), please convert your local time to UTC.

Reviewer 3: I'm fine if this doesn't get changed, but we state `UTC (24-hour time notation)` a few places now, which makes it sound like it is just a date format.

I would hope most people are aware of UTC, but it would be good the first time to expand it out to the full "Coordinated Universal Time" instead of just the acronym, and rather than saying "24-hour time notation" each time, provide a link to something like [a time converter](https://www.timeanddate.com/worldclock/converter.html?iso=20230124T060000&p1=1440) to help the user figure out the UTC time for their desired window.

Author: That makes sense to me! I can provide a more detail description about UTC and guide users to figure out the UTC time zone.

# | ┌───────────── minute (0 - 59)
# | │ ┌───────────── hour (0 - 23)
# | │ │ ┌───────────── day of the month (1 - 31)
# | │ │ │ ┌───────────── month (Jan, Feb, Mar, Apr, Jun, Jul, Aug, Sep, Oct, Nov, Dec)
# | │ │ │ │ ┌───────────── day of the week (Mon, Tue, Wed, Thu, Fri, Sat, Sun)
# | │ │ │ │ │ ┌───────────── year (formatted as YYYY)
# | │ │ │ │ │ |
# | │ │ │ │ │ |
# * * * * * * *
```

> Note: brupop uses Coordinated Universal Time(UTC), please convert your local time to Coordinated Universal Time(UTC). This tool [Time Zone Converter](https://www.timeanddate.com/worldclock/converter.html) can help you find your desired time window on UTC.

Reviewer 2: Nit:
```
Coordinated Universal Time (UTC)
```

Should have a space between the definition and the abbreviation.
